services/ui-src/src/utils/state/useStore.ts

```diff
@@ -99,24 +98,22 @@ const reportStore = (set: Function) => ({
 // ENTITY STORE
 const entityStore = (set: Function) => ({
   // initial state
-  entityId: undefined,
-  entityType: undefined,
-  entities: [],
   selectedEntity: undefined,
+  entityType: undefined,
   // actions
-  setEntityType: (newEntityType: EntityType | undefined) =>
-    set(() => ({ entityType: newEntityType }), false, {
-      type: "setEntityType",
-    }),
-  setEntities: (newEntities: EntityShape[] | undefined) =>
-    set(() => ({ entities: newEntities }), false, {
-      type: "setEntities",
-    }),
-  clearEntities: () =>
-    set(() => ({ entities: [] }), false, { type: "clearEntities" }),
   setSelectedEntity: (newSelectedEntity: EntityShape | undefined) =>
-    set(() => ({ selectedEntity: newSelectedEntity }), false, {
-      type: "setSelectedEntity",
+    set(
+      () => ({
+        selectedEntity: newSelectedEntity,
+      }),
+      false,
+      {
+        type: "setSelectedEntity",
+      }
+    ),
+  clearSelectedEntity: () =>
+    set(() => ({ selectedEntity: undefined }), false, {
+      type: "clearSelectedEntity",
```

Comment:
cleaned up store for entities, so it more accurately reflects data that is needed + makes it easier to utilize. entityType and entityId have been moved to live inside selectedEntity. Also added clearSelectedEntity()